src/components/earthengine/Collection.js

```diff
@@ -51,31 +51,39 @@ export class CollectionSelect extends Component {
 
         if (id && !collections[id]) {
             loadEarthEngineCollection(id);
+        } else {
+            this.setYears();
         }
     }
 
     componentDidUpdate(prevProps) {
-        const { id, filter, collections } = this.props;
+        const { id, collections } = this.props;
 
         if (id && collections[id] !== prevProps.collections[id]) {
```

Comment:
For completeness should this be `if (id && collections && collections[id] && collections[id] !== prevProps.collections[id])` ?  Will `collections[id]` ever **become** null?